src/progress/progress_set.rs

```diff
@@ -544,12 +444,7 @@ impl ProgressSet {
             }
             pr.recent_active = false;
         }
-        for (&_id, pr) in self.learners_mut() {
```

Comment:
Why skip setting them to false?

Reply:
`recent_active` is only used for checking quorum on leader. So `quorum_recently_active` will never care about leaners. You can find the same logic in etcd:
```go
// QuorumActive returns true if the quorum is active from the view of the local
// raft state machine. Otherwise, it returns false.
func (p *ProgressTracker) QuorumActive() bool {
    votes := map[uint64]bool{}
    p.Visit(func(id uint64, pr *Progress) {
        if pr.IsLearner {
            return
        }
        votes[id] = pr.RecentActive
    })
    return p.Voters.VoteResult(votes) == quorum.VoteWon
}
```
And
```go
        if !r.prs.QuorumActive() {
            r.logger.Warningf("%x stepped down to follower since quorum is not active", r.id)
            r.becomeFollower(r.Term, None)
        }
        // Mark everyone (but ourselves) as inactive in preparation for the next
        // CheckQuorum.
        r.prs.Visit(func(id uint64, pr *tracker.Progress) {
            if id != r.id {
                pr.RecentActive = false
            }
        })
```

Reply:
Oh, I find the mistake. Learners still to be marked in the second sector.

Reply:
Fixed.